src/neural_network/layer.py

```diff
@@ -28,3 +29,6 @@ def update_weights_and_biases(self, nabla_w, nabla_b, eta, mini_batch_size, mome
         self.bias_velocities = momentum * self.bias_velocities - (eta / mini_batch_size) * nabla_b
         self.weights = self.weights + self.weight_velocities
         self.biases = self.biases + self.bias_velocities
+
+    def add_weights_to_history(self):
+        self.weights_history.append(self.weights)
```

Comment:
Tę linijkę przerzuciłabym na początek funkcji `update_weights_and_biases`, wtedy można usunąć funkcję `add_weights_to_history`

Reply:
Jeśli przerzucilibyśmy ją do `update_weights_and_biases`, to wtedy zapisywalibyśmy w historii wagi krawędzi po przepuszczeniu każdego pojedynczego inputu, a do wykresów potrzebujemy wag z końca każdej epoki. Dlatego właśnie dodałem to jako funkcję, żeby mogła być wywołana z poziomu klasy sieci w czasie uczenia na końcu każdej epoki